client/src/pages/private/Admin.js

```diff
@@ -51,33 +53,26 @@ export default () => {
 
   return (
     <Page >
-      <Grid container alignContent="center" justify="center" alignItems="center" direction="column">
-        <Box pb={4} pl={4} pr={4} pt={2}>
-          <Grid container direction="column">
-            {
-              roles.length === 0 ?
-                <Typography variant="subtitle1" gutterBottom>
-                  You don't have enough permissions.
+      <CheckPermissions roles={roles} permittedRoles={['maximus', 'employer', 'health_authority', 'ministry_of_health']} renderMessage={true}>
+        <Grid container alignContent="center" justify="center" alignItems="center" direction="column">
+          <Box pb={4} pl={4} pr={4} pt={2}>
+            <Grid container direction="column">
+              <Typography variant="subtitle1" gutterBottom>
+                Welcome, {name}
               </Typography>
-                :
-                roles.includes('admin') ?
-                  [
-                    renderAdminButton(0, Routes.ApplicantUpload, 'Upload Applicants'),
-                    renderAdminButton(1, Routes.EOIView, 'Expressions of Interest'),
-                  ]
-                  :
-                  roles.map((item, index) => {
-                    switch (item) {
-                      case 'maximus':
-                        return renderAdminButton(index, Routes.ApplicantUpload, 'Upload Applicants');
-                      default:
-                        return null
-                    }
-                  })
-            }
-          </Grid>
-        </Box>
-      </Grid>
+              <CheckPermissions roles={roles} permittedRoles={['maximus']}>
+                { renderAdminButton(0, Routes.ApplicantUpload, 'Upload Applicants') }
```

Comment:
I'm not 100% sure, but I think this first parameter is no longer necessary since these components have their own parent now.

Reply:
Looks correct from my testing :) thanks!